up-l1/mqtt.adoc

```diff
@@ -35,8 +35,126 @@ MQTT is an OASIS standard messaging protocol for the Internet of Things (IoT). I
 
 For more information, please refer to https://mqtt.org/
 
-
 == Specifications
 
+=== UAttribute Mapping
+
+MQTT 5 supports custom header fields, and we leverage this to map the UAttribute values into the MQTT header.
+
+[cols="1,1"]
+|===
+| MQTT Header Property Code | Value (key:value)
+| UserProperty
+| ("1", "{UAttribute.id}") # UUID id
+| UserProperty
+| ("2", "{UAttribute.type}") # UMessageType
+| UserProperty
+| ("3","{UAttribute.source}") # Short Form Serialized Source UURI
+| UserProperty
+| ("4","{UAttribute.sink}") # Short Form Serialized Sink UURI
+| UserProperty
+| ("5","{UAttribute.priority}") # UPriority
+| UserProperty
+| ("6", "{UAttribute.ttl}") # Optional ttl
+| UserProperty
+| ("7", "{UAttribute.permissionLevel}") # Optional permission level
+| UserProperty
+| ("8", "{UAttribute.commStatus}") # Optional UCode commstatus
+| UserProperty
+| ("9", "{UAttribute.reqId}") # UUID reqid
+| UserProperty
+| ("10", "{UAttribute.token}") # Optional token
+| UserProperty
+| ("11", "{UAttribute.traceParent}") # Optional traceparent
+|===
+
+All uAttributes are mapped to a MQTT header UserProperty, where the key is the UAttribute protobuf field number.
+
+The source and sink fields **MUST** be link:../basics/uri.adoc[Short URI Format].
+
+The type field is used by a UListener to select message types on the topic.
+
+=== URI Mapping
+
+The URI mapping follows the short form UUri format, with the addition of a location identifier to determine if the message is intended for a local or cloud destination.
+
+- `{UUri.UAuthority.number}/{UUri.UEntity.id}/{UUri.UEntity.version_major}/{UUri.UResource.id}`
+
+  Note: If the UUri does not contain a UAuthority, add the UAuthority.number for the UAuthority the transport client is running on.
+
+Subscribe to all messages to the current Authority
+
+- `{UUri.UAuthority.number}/+/+/+` where `UAttributes.type != Publish`
```

Comment:
Filtering based on UserProperty is not inherently supported by MQTT's native subscribe function. Instead, this type of filtering have to be implemented at the application level.  
whether it's the cloud subscribing to messages from a vehicle or a vehicle subscribing to cloud messages, this approach is inefficient.  specially on Cloud side.

Reply:
Hmm, what do you think about appending the message type to the topic?

Reply:
> Filtering based on UserProperty is not inherently supported by MQTT's native subscribe function. Instead, this type of filtering have to be implemented at the application level. whether it's the cloud subscribing to messages from a vehicle or a vehicle subscribing to cloud messages, this approach is inefficient. specially on Cloud side.

@lynn2009 Right now the purpose of this is only for D2C and C2D and once the message is in the cloud, it will use the clouse native uTransport and uSubscription infrastructure (i.e. Eventhub or something else).

Reply:
> Hmm, what do you think about appending the message type to the topic?

Or maybe even put it to the front as the most discriminant field.
Filtering on all Publish messages: `pub.v1/+/+/+/+`
Filtering on all messages destined to myself: `+/{my.authority.id}/+/+/+`

Reply:
Prefer to have the following patterns:
rpc/{authority}/+/+/+
notf/{authority}/+/+/+
pub/{authority}/+/+/+

This approach enables subscribers to subscribe without needing application-level filtering, making it simpler to implement authorization controls. For instance, the uAuthority would be restricted to subscribing only to topics under rpc/<UUri.uAuthority.number|name>/# and notf/<UUri.uAuthority.number|name>/#, but they would be allowed to publish to topics under pub/<UUri.uAuthority.number|name>/#.